lib/mongo/bulk_write/transformable.rb

```diff
@@ -90,6 +90,7 @@ module Transformable
           Operation::UPSERT => doc.fetch(:upsert, false)
         }.tap do |d|
           d[Operation::COLLATION] = doc[:collation] if doc[:collation]
+          d[Operation::ARRAY_FILTERS] = doc[:array_filters] if doc[:array_filters]
```

Comment:
Sorry, this was my mistake to tell you to merge in array_filters here. Array filters aren't relevant to a replace_one, right? If not, you don't need this line. That was my error, sorry!